tests/components/homeassistant_sky_connect/test_init.py

```diff
@@ -0,0 +1,104 @@
+
+import pytest
+
+from homeassistant.components.homeassistant_sky_connect.const import DOMAIN
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+
+CONFIG_ENTRY_DATA = {
+    "device": "bla_device",
+    "vid": "bla_vid",
+    "pid": "bla_pid",
+    "serial_number": "bla_serial_number",
+    "manufacturer": "bla_manufacturer",
+    "description": "bla_description",
+}
+
+
+@pytest.mark.parametrize(
+    "onboarded, num_entries, num_flows", ((False, 1, 0), (True, 0, 1))
+)
+async def test_setup_entry(
+    hass: HomeAssistant, onboarded, num_entries, num_flows
+) -> None:
+    """Test setup of a config entry, including setup of zha."""
+    # Setup the config entry
+    config_entry = MockConfigEntry(
+        data=CONFIG_ENTRY_DATA,
+        domain=DOMAIN,
+        options={},
+        title="Home Assistant Sky Connect",
+    )
+    config_entry.add_to_hass(hass)
+    with patch(
+        "homeassistant.components.homeassistant_sky_connect.usb.async_is_plugged_in",
+        return_value=True,
+    ) as mock_is_plugged_in, patch(
+        "homeassistant.components.onboarding.async_is_onboarded", return_value=onboarded
+    ), patch(
+        "zigpy_znp.zigbee.application.ControllerApplication.probe", return_value=True
+    ):
+        assert await hass.config_entries.async_setup(config_entry.entry_id)
+        await hass.async_block_till_done()
+        assert len(mock_is_plugged_in.mock_calls) == 1
+
+    assert len(hass.config_entries.async_entries("zha")) == num_entries
+    assert len(hass.config_entries.flow.async_progress_by_handler("zha")) == num_flows
+
+
+async def test_setup_zha(hass: HomeAssistant) -> None:
+    """Test zha gets the right config."""
+    # Setup the config entry
+    config_entry = MockConfigEntry(
+        data=CONFIG_ENTRY_DATA,
+        domain=DOMAIN,
+        options={},
+        title="Home Assistant Sky Connect",
+    )
+    config_entry.add_to_hass(hass)
+    with patch(
+        "homeassistant.components.homeassistant_sky_connect.usb.async_is_plugged_in",
+        return_value=True,
+    ) as mock_is_plugged_in, patch(
+        "homeassistant.components.onboarding.async_is_onboarded", return_value=False
+    ), patch(
+        "zigpy_znp.zigbee.application.ControllerApplication.probe", return_value=True
+    ):
+        assert await hass.config_entries.async_setup(config_entry.entry_id)
+        await hass.async_block_till_done()
+        assert len(mock_is_plugged_in.mock_calls) == 1
+
+    config_entry = hass.config_entries.async_entries("zha")[0]
+    assert config_entry.data == {
+        "device": {"baudrate": 115200, "flow_control": None, "path": "bla_device"},
+        "radio_type": "znp",
+    }
+    assert config_entry.options == {}
+    assert (
+        config_entry.title
+        == "bla_description - bla_device, s/n: bla_serial_number - bla_manufacturer - bla_vid:bla_pid"
+    )
```

Comment:
This should perhaps be customized for sky connect dongles?

Reply:
Why ever name the entry this crazy text haha

Reply:
Usually we don't know what it is, but in this case we could probably give it a better name

Reply:
Sure. But why not just use the description and if that is empty or null then don't set it 🤷

Reply:
The data is all over the place for various vendors.  You can end up with lots of adapters that say 'USB' or 'Serial' without any way to tell them apart and there doesn't seem to be any standard on what to put in each field. The text fields seem to have any old random data shoved into them as it's all vendor specified

Reply:
Then dont pump anything at all. Better to just say Zigbee

Reply:
That probably works fine.  I don't think a lot of people have more than one

Reply:
I dont think you can have more than 1 zigbee integration right?

Reply:
I have Deconz and ZHA and Zigbee2MQTT all running with 3 different dongles for testing.. but thats probably a horrible idea for production

Reply:
ZHA only supports a single config entry, so if anything the config entry should be named zha + possibly some short name identifying the dongle 🤷‍♂️

Reply:
This simplifies the title: https://github.com/home-assistant/core/pull/76991